### DISCUSSION TITLE
How to paginate a document subfield

### DISCUSSION THREAD

**Discussion by kheyer** [BeanieODM/beanie#716] [Question]
I have a document model

```python
class MyDocument(Document):
    subfield: List
```

`MyDocument.subfield` might have a huge number of entries. When querying, I want to paginate through `MyDocument.subfield` with `skip` and `limit` terms. How can I do this? They query in this case would specify the `id` of the document in question, as well as `skip` and `limit` values.

Something like `MyDocument.get(doc_id).subfield.skip(skip).limit(limit).to_list()`. Is there a way to do this with a projection or the `$slice` operator?

**Comment by mrrobinhood5** [comment]
You can do it in aggregation. heres the js used to accomplish something like this:
```js
db.getCollection("TeamBase").aggregate([
    {'$match': {'_id': ObjectId("64fbf0de348c944c4a50736c")}},
    {'$addFields': {'members':{'$slice': ['$members', 2]}}}
])
```

In Beanie: 
```py
one = PydanticObjectId("64fbf0de348c944c4a50736c")
results = await (TeamBase.find(TeamBase.id == one, with_children=True)
                    .aggregate([{'$addFields': {'members':{'$slice': ['$members', 2]}}}], projection_model=TeamBase)
                    .to_list())
```

or you can optionaly use Monggregate's Pipeline:

```py
one = PydanticObjectId("64fbf0de348c944c4a50736c")
pipeline = Pipeline().match({'_id': one}).add_fields({'members':{'$slice':['$members', 2]}}).export()
results = await TeamBase.find({}, with_children=True).aggregate(pipeline, projection_model=TeamBase).to_list()
```
